core/src/main/java/org/kiwix/kiwixmobile/core/extensions/ActivityExtensions.kt

```diff
@@ -81,27 +83,51 @@ object ActivityExtensions {
       .get(T::class.java)
 
   fun Activity.navigate(action: NavDirections) {
-    (this as CoreMainActivity).navigate(action)
+    asCoreMainActivity().navigate(action)
   }
 
   val Activity.cachedComponent: CoreActivityComponent
-    get() = (this as CoreMainActivity).cachedComponent
+    get() = asCoreMainActivity().cachedComponent
 
   fun Activity.setupDrawerToggle(toolbar: Toolbar) =
-    (this as CoreMainActivity).setupDrawerToggle(toolbar)
+    asCoreMainActivity().setupDrawerToggle(toolbar)
 
   fun Activity.navigate(fragmentId: Int) {
-    (this as CoreMainActivity).navigate(fragmentId)
+    asCoreMainActivity().navigate(fragmentId)
   }
 
   fun Activity.navigate(fragmentId: Int, bundle: Bundle) {
-    (this as CoreMainActivity).navigate(fragmentId, bundle)
+    asCoreMainActivity().navigate(fragmentId, bundle)
   }
 
   fun Activity.popNavigationBackstack() {
-    (this as CoreMainActivity).navController.popBackStack()
+    asCoreMainActivity().navController.popBackStack()
   }
 
-  val Activity.coreActivityComponent
-    get() = CoreApp.coreComponent.activityComponentBuilder().activity(this).build()
+  private fun Activity.asCoreMainActivity() = (this as CoreMainActivity)
```

Comment:
maybe this can be a property `coreMainActivity` instead

Reply:
Clean suggestion!